### ISSUE TITLE
Send mails works only when "outgoing" folder is active

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#538]
**Originally by _stellan.pistoor@gmx.de_ on 2014-03-19 17:14:27 +0100**

___
If not the "outgoing" folder is active send mails doesn`t work (no progress window appears, mail(s) isn`t sent). When I select the "outgoing" folder it works as expected.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-03-19 17:26:13 +0100**

___
Correct myself. It seems that something else is resposible for "not send mails erro". Maybe some MUI stuff. Have to test more.

**Comment by tboeckel** [comment]
**Originally on 2014-03-19 19:18:17 +0100**

___
Works perfectly here, no matter which folder is active. Are you sure there is a mail in the Outgoing folder to be sent at all?

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-03-20 13:48:44 +0100**

___
Yes, I`m sure there was one email in the outgoing folder. After changing (activate) from incoming to outgoing folder and click on "Send" the email was sent. But as I wrote there is somthing other because it seems it happens only sometimes. So far you can close this issue. If I have something that I can reproduce I`ll reopen it.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-03-20 17:54:07 +0100**

___
Some littel more information. I just want to send a mail but nothing happens after click on "Send" as mentioned above. What I did:

- Clicked on outgoing folder
- Then back (click) to incoming folder
- Now click on "Send"

Result is that YAM start (tries because of my SSL erros) to send the mail. Ok, mail wasn`t sent because of SSL error. Now:

- Click on "Send" again

Result: Nothing happens

So, something reset while doing this. I `ve to have to at least outgoing folder once active before I can send a mail.

I hope the description is clear enough and help a bit.

**Comment by tboeckel** [comment]
**Originally on 2014-03-20 23:14:21 +0100**

___
I just added some more debug output to the mail sending preparation. Please try to reproduce this issue with the debug version of the next nightly build and create a MAIL debug log with it.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-04-06 22:08:12 +0200**

___
Mail debug log attached. When I hit "Send" button the mail in outgoing folder isn`t sent and there is nothing add to the log while doing this. When outgoing folder is active the mail will sent.

It seems if two folders (not outgoing) are selected = send mail doesn`t work.
When outgoing folder is active then select e.g. incoming folder mail is send.

**Comment by tboeckel** [comment]
**Originally on 2014-04-22 14:48:07 +0200**

___
Sorry for the long delay. Unfortunately the debug log does not yet contain the information I am interested in. All I can tell so far is that the send process is indeed started, but there is no indication why the mail is not sent.

Please a new MAIL + NET debug log with the next nightly build as I have just fixed a broken debug statement.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-05-02 14:08:37 +0200**

___
**Attachment added:** [`Yam_Send_Mail.txt`](https://yam.ch/attachments/538/Yam_Send_Mail.txt) (22.5 KiB)

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-05-02 14:08:37 +0200**

___
Attachment (Yam_Send_Mail.txt) added by ticket reporter.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-05-02 14:10:54 +0200**

___
New log attached. One mail was in outgoing folder. Active was incoming folder. Click several times on "Send" button does nothing. After click on outgoing folder and "Send" button the mail was sent.

**Comment by tboeckel** [comment]
**Originally on 2014-05-03 12:39:11 +0200**

___
In (f8133bc):
* YAM_MA.c: added some more debug output to track down the send mail issue. This refs #538.

**Comment by tboeckel** [comment]
**Originally on 2014-05-03 12:43:05 +0200**

___
In (748ee40):
* YAM_MA.c: added the mode value to the debug output. This refs #538.

**Comment by tboeckel** [comment]
**Originally on 2014-05-03 12:59:06 +0200**

___
I have added some more debug output. Please create yet another MAIL + NET log with the next nightly build.

**Comment by tboeckel** [comment]
**Originally on 2014-05-03 13:18:03 +0200**

___
In (af76004):
* YAM_MA.c: added even more debug output for ticket #538.

**Comment by tboeckel** [comment]
**Originally on 2014-05-08 21:49:45 +0200**

___
Bump!

**Comment by jens-maus** [comment]
**Originally by _trac-robot_ on 2014-05-23 07:10:50 +0200**

___
This ticket was closed automatically by the system.  It was previously set to a Pending status and hasn't been updated within 14 days.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-05-23 17:11:41 +0200**

___
Since a week sending mails completle fails while SSL stuff (maybe gmx changed something?). Further, cannot dowload new debug version. So, this issue have to wait. Sorry!

Following hasn`t (more or less) something to do with this bugreport but just an explanation if you are interested:

```
** YAM 2.10-dev [OS3/m68k, r7826, GCC 2.95.x] date 27.04.2014 (build 20140427) startup **********************
Exec version: v40.10
Initializing runtime debugging:
parsing ENV:yamdebug content '!startup mail net'
clear 'startup' debug flag
set 'mail' debug flag
set 'net' debug flag
set debug classes/flags (env:yamdebug): 000000F6/00000441
set debug files: ''
** Normal processing follows ***************************************
[...]
01:D:   smtp.c:1455:prepared 1 mails for sending, 3207 bytes
[...]
01:D:    Connection.c:1263:connection to 'mail.gmx.net:465' succeeded
01:D:    ssl.c:711:CAfile # 'PROGDIR:Resources/certificates/ca-bundle.crt', CApath'PROGDIR:Resources/certificates'
01:D:    ssl.c:744:initializing TLS/SSL session
01:D:    ssl.c:758:available SSL ciphers:
01:D:    ssl.c:762:DHE-RSA-AES256-SHA
01:D:    ssl.c:762:DHE-DSS-AES256-SHA
01:D:    ssl.c:762:AES256-SHA
01:D:    ssl.c:762:EDH-RSA-DES-CBC3-SHA
01:D:    ssl.c:762:EDH-DSS-DES-CBC3-SHA
01:D:    ssl.c:762:DES-CBC3-SHA
01:D:    ssl.c:762:DHE-RSA-AES128-SHA
01:D:    ssl.c:762:DHE-DSS-AES128-SHA
01:D:    ssl.c:762:AES128-SHA
01:D:    ssl.c:762:DHE-DSS-RC4-SHA
01:D:    ssl.c:762:RC4-SHA
01:D:    ssl.c:762:RC4-MD5
01:D:    ssl.c:771:set socket descriptor 0 for context 023082fc
01:D:    ssl.c:780:connect SSL context 023082fc
01:C:     ssl.c:181:x509_ctx = 27463676, 0x01A30FFC
01:C:     ssl.c:182:parm = 30602808, 0x01D2F638
01:C:      ssl.c:92:preverify_ok = 1, 0x00000001
01:C:      ssl.c:93:x509_ctx = 27463676, 0x01A30FFC
01:C:      ssl.c:92:preverify_ok = 1, 0x00000001
01:C:      ssl.c:93:x509_ctx = 27463676, 0x01A30FFC
01:C:      ssl.c:92:preverify_ok = 1, 0x00000001
01:C:      ssl.c:93:x509_ctx = 27463676, 0x01A30FFC
```

At this point nothing happens for minutes and task "Yam thread [0]" takes all available cpu time (means 100% cpu usage). While this I can use IBrowse but slow of course. Normaly I geht the SSL error message sooner or in case of no error mail was just sent. If I must guessing then I would say GMX has changed something.

```
01:E:    ssl.c:842:SSL_connect() returned 0 with SSL_get_errror() = 5
01:E:    ssl.c:853:errno(22) = 's£'
01:E:    ssl.c:855:errnosv(0) = 's£'
01:E:    ssl.c:861:querying ERR_get_error() stack:
[...]
```

**Comment by tboeckel** [comment]
**Originally on 2014-05-23 21:51:52 +0200**

___
Replying to stellan:
> Since a week sending mails completle fails while SSL stuff (maybe gmx changed something?). Further, cannot dowload new debug version. So, this issue have to wait. Sorry!

GMX did not change anything from my point of view. Everything is working perfectly with my own GMX account. And why are you unable download the nightly builds if you are able comment here?

> Following hasn`t (more or less) something to do with this bugreport but just an explanation if you are interested:

This has nothing to do with this ticket and does not belong here. If at all it is ticket #530 which covers SSL issues.

> ** YAM 2.10-dev [OS3/m68k, r7826, GCC 2.95.x] date 27.04.2014 (build 20140427) startup **********************

Trying to prove anything with a 4 weeks old nightly build is a bad idea, especially since Jens changed some things regarding SSL certificate handling recently.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-05-25 20:01:59 +0200**

___
New NET MAIL log attached using 2.10-dev [OS3/m68k] (25.05.14). While incoming folder is active and I click on "Send" this log entry is made:

00:W:  YAM_MA.c:2727:failed to create mail list for folder 'Outgoing'

When I select outgoing folder and click "Send" log goes on.

**Comment by tboeckel** [comment]
**Originally on 2014-05-25 21:29:17 +0200**

___
Sorry, but this is no NET+MAIL log, but just a simple NET log. There is definitely a message missing before the failed creation of the mail list. Please create it again and **do not** strip anything.

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-05-26 16:25:52 +0200**

___
**Attachment added:** [`Yam_Send_Mail3.txt`](https://yam.ch/attachments/538/Yam_Send_Mail3.txt) (11.4 KiB)
new

**Comment by jens-maus** [comment]
**Originally by _stellan.pistoor@gmx.de_ on 2014-05-26 16:27:48 +0200**

___
I also wondered when I looked at the log. New log attached but MAIL stuff isn`t there.

**Comment by tboeckel** [comment]
**Originally on 2014-05-26 20:00:54 +0200**

___
Ok, finally I was able to reproduce this issue myself. The reason is quite simple: the Outgoing folder's index was not loaded or flushed and hence there was no list of mails to be cloned. This is easy to fix.

**Comment by tboeckel** [comment]
**Originally on 2014-05-26 20:02:41 +0200**

___
In (3fceaf1):
* YAM_MA.c: make sure to load the Outgoing folder's index before trying to clone its list of mails for sending. This closes #538.

**Comment by tboeckel** [comment]
**Originally on 2014-05-26 21:11:00 +0200**

___
In (fbec876):
* misc: make sure that the folder's index is loaded and valid in some more situations. This refs #538.
